lua/luasnip/nodes/functionNode.lua

```diff
@@ -1,7 +1,19 @@
 local FunctionNode = require("luasnip.nodes.node").Node:new()
 local util = require("luasnip.util.util")
+local lambda = require("luasnip.util.lambda")
 
 local function F(fn, args, ...)
+	if lambda.isPE(fn) then
+		local function _concat(tbl)
+			return table.concat(tbl, "\n")
+		end
+		local expr = lambda.instantiate(fn)
+		fn = function(args)
+			local inputs = vim.tbl_map(_concat, args)
+			local out = expr(unpack(inputs))
+			return vim.split(out, "\n")
```

Comment:
Nice, but wouldn't a more concise `return vim.split(expr(table.concat(args[1], "\n")), "\n")` also do it? These function nodes depend on only one other node, therefore args would always be `{{"line 1", "maybe line 2"}}`.

Reply:
It can be more than one node,  as in   `_1 .. _2` , or   `_1:gsub(_2)`

Reply:
Anyway, I had to stop for today, but tomorrow morning I will move it to a function out of the function_node as we talk before.

Reply:
Oh wow, that's even better, I didn't realize the lambdas were this powerful :0
Sounds good, thank you!!